rstudio/rstudio
# remove old Makevars file generated by RStudio

### Intent

Previously, RStudio wrote a file at `~/.R/Makevars` on macOS, to help users manage the transition from gcc to clang (IIRC this process began with Snow Leopard). Unfortunately, this file conflicts with the way R sets the default compiler flags now, since the default compiler flags are encoded as part of the `CXX` make variable.

This PR then removes the RStudio-generated `~/.R/Makevars` if appropriate.

Addresses https://github.com/rstudio/rstudio/issues/8800.

### Approach

Straightforward removal of `~/.R/Makevars` if its contents matches what we previously wrote to it.

### QA Notes

With R 4.0.3 on macOS, first execute the following:

```
dir.create("~/.R", showWarnings = FALSE)
writeLines("CC=clang\nCXX=clang++", "~/.R/Makevars")
R <- file.path(R.home("bin"), "R")
system(paste(R, "CMD config CXX"))
```

Check that the following is reported:

```
> R <- file.path(R.home("bin"), "R")
> system(paste(R, "CMD config CXX"))
clang++
```

Then, restart the R session, and re-run this part of the code:

```
R <- file.path(R.home("bin"), "R")
system(paste(R, "CMD config CXX"))
```

You should see something like:

```
> R <- file.path(R.home("bin"), "R")
> system(paste(R, "CMD config CXX"))
clang++ -mmacosx-version-min=10.13 -std=gnu++11
```

and the file at `~/.R/Makevars` should no longer exist.
